### PR TITLE
Shuffle Finalizer Tests + bug fix

### PR DESCRIPTION
**Pull request by benthecarman**
After trying to implement the ShuffleFinalizer in the wallet I found some bugs, I fixed them and added tests to make sure that doesn't happen again!

Also found a bug in `isValidFeeRange` where if `estimatedFee` and `actualFee` were both `0` it would fail, so I fixed that as well.

### CODE REVIEW

File: core-test/src/test/scala/org/bitcoins/core/wallet/builder/RawTxFinalizerTest.scala [hunk 1]
```diff
@@ -131,6 +140,54 @@ class RawTxFinalizerTest extends BitcoinSAsyncTest {
...
+        val txBuilder = RawTxBuilder() ++= inputs ++= outputs
+        val finalized = txBuilder.setFinalizer(ShuffleInputsFinalizer)
+        val txsF =
+          FutureUtil.foldLeftAsync(Vector.empty[Transaction], 0 to 20) {
+            (accum, _) => finalized.buildTx().map(_ +: accum)
+          }
+        txsF.map(txs =>
+          assert(
+            inputs.size <= 1 || txs.exists(
+              _.inputs.map(_.previousOutput) != inputs.map(_.outPoint))))
```
Comment (nkohen) [inline core-test/src/test/scala/org/bitcoins/core/wallet/builder/RawTxFinalizerTest.scala:155]: If the number of inputs is small, won't this assertion have a decent chance of failing?

File: core-test/src/test/scala/org/bitcoins/core/wallet/builder/RawTxFinalizerTest.scala [hunk 2]
```diff
@@ -131,6 +140,54 @@ class RawTxFinalizerTest extends BitcoinSAsyncTest {
...
+    forAllAsync(CreditingTxGen.inputsAndOutputs()) {
+      case (inputs, outputs) =>
+        val txBuilder = RawTxBuilder() ++= inputs ++= outputs
+        val finalized = txBuilder.setFinalizer(ShuffleOutputsFinalizer)
+        val txsF =
+          FutureUtil.foldLeftAsync(Vector.empty[Transaction], 0 to 20) {
+            (accum, _) => finalized.buildTx().map(_ +: accum)
+          }
+        txsF.map(txs =>
+          assert(outputs.size <= 1 || txs.exists(_.outputs != outputs)))
```
Comment (nkohen) [inline core-test/src/test/scala/org/bitcoins/core/wallet/builder/RawTxFinalizerTest.scala:169]: diito

File: core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala [hunk 3]
```diff
@@ -319,19 +319,25 @@ case class ShufflingNonInteractiveFinalizer(
...
-      .buildTx(txBuilderResult)
+      val shuffledInputInfos = tempTxBuilderResult.inputs
+        .map(input => inputInfos.find(_.outPoint == input.previousOutput).get)
+
+      val sanityCheck =
+        SanityCheckFinalizer(inputInfos = shuffledInputInfos,
+                             expectedOutputSPKs =
+                               tempTxBuilderResult.outputs.map(_.scriptPubKey),
+                             expectedFeeRate = feeRate,
+                             changeSPKs = Vector(changeSPK))
```
Comment (nkohen) [inline core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala]: Why did this change need to happen? Do any of the sanity checks depend on order?

File: core-test/src/test/scala/org/bitcoins/core/wallet/builder/RawTxFinalizerTest.scala (hunk 1 above)
Comment (benthecarman) [reply core-test/src/test/scala/org/bitcoins/core/wallet/builder/RawTxFinalizerTest.scala:155]: that's why we generate 20 of them

File: core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala (hunk 3 above)
Comment (benthecarman) [reply core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala]: Fixed, turns out don't need this after the bug fix

**Pull request merged**